docs/designs/template-000.md

```diff
@@ -0,0 +1,9 @@
+# Title
+
+## Overview
+
+## Context
```

Comment:
"Overview" should serve the same function as an abstract in a paper: to give a brief summary of the main body of text. It is necessary because it provides means to skip or postpone reading of the entire doc. I think naming could be discussed, but having a TL;DR section is important.

Reply:
Then it should be called abstract, because I think the definition of abstract is much stricter than that of an overview, what we're doing with these design docs wrt/ pkg manager background is also an overview.
The question still remains, should an abstract be mandatory for the original design docs? I can imagine them being mandatory for the obsolete/update docs, but not for the main document, the title kinda already hints at what it's about --> implemeting new backend.

Reply:
Mandatory fields mean that these fields are applicable to all designs, determining a set of universally applicable ones would require significant effort to analyze all existing designs and distill out all common parts. There is also a big chance of introducing too many unnecessary fields. I've seen mandatory templates with more than half fields ending up being marked as "Non-applicable" which created lots of visual noise and made it easy to miss something important. In some cases it was more productive not to follow the template at all. I don't think it is a good idea to set rules which will not be followed (for whatever reason). I am also concerned with extra maintenance and cognitive load which mandatory fields introduce so would suggest keeping them to a reasonable minimum.

Reply:
@a-ovchinnikov I agree with that reasoning so as long as we don't end up with all fields optional and their intent is clearly defined, we're good.